google/cloud/storage/internal/lifecycle_rule_parser.cc

```diff
@@ -20,6 +20,51 @@ namespace cloud {
 namespace storage {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace internal {
+namespace {
+
+absl::optional<std::vector<std::string>> ParseStringListCondition(
+    nlohmann::json const& condition, char const* name) {
+  if (!condition.contains(name)) return absl::nullopt;
+  std::vector<std::string> matches;
+  for (auto const& kv : condition[name].items()) {
+    matches.emplace_back(kv.value().get<std::string>());
+  }
+  return matches;
+}
+
+Status ParseIntCondition(absl::optional<std::int32_t>& field,
+                         nlohmann::json const& condition, char const* name) {
+  if (!condition.contains(name)) return Status{};
+  auto value = internal::ParseIntField(condition, name);
+  if (!value) return std::move(value).status();
+  field.emplace(*value);
+  return Status{};
+}
+
+Status ParseBoolCondition(absl::optional<bool>& field,
+                          nlohmann::json const& condition, char const* name) {
+  if (!condition.contains(name)) return Status{};
+  auto value = internal::ParseBoolField(condition, name);
+  if (!value) return std::move(value).status();
+  field.emplace(*value);
+  return Status{};
+}
+
+Status ParseDateCondition(absl::optional<absl::CivilDay>& field,
+                          nlohmann::json const& condition, char const* name) {
+  if (!condition.contains(name)) return Status{};
+  auto const date = condition.value(name, "");
+  absl::CivilDay day;
+  if (!absl::ParseCivilTime(date, &day)) {
+    return Status(StatusCode::kInvalidArgument,
+                  "Cannot parse createdBefore value (" + date + ") as a date");
```

Comment:
Use `name` instead of literal "createdBefore"?

Reply:
Did you forget to push?

Reply:
Doh!  Done.